File: frontend/lib/model/donation.dart
import 'package:frontend/model/status_donation.dart';

import 'donation_item.dart';

class Donation {
  final int? id;
  final String title;
  final String description;
  final StatusDonation status;
  final DateTime createdAt;
  final DateTime updatedAt;
  final List<DonationItem> items;
  final int? donorId;

  Donation({
    this.id,
    required this.title,
    required this.description,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
    required this.items,
    required this.donorId
  });

  factory Donation.fromJson(Map<String, dynamic> json) {
    return Donation(
      id: json['id'],
      title: json['title'],
      donorId: json['donorId'],
      description: json['description'],
      status: StatusDonation.fromString(json['status']),
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
      items: (json['items'] as List<dynamic>)
          .map((e) => DonationItem.fromJson(e))
          .toList(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'donorId': donorId,
      'status': status.name,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
      'items': items.map((i) => i.toJson()).toList(),
    };
  }
}

File: backend/src/main/kotlin/br/com/helpiguacu/backend/infrastructure/persistence/donation/Donation.kt
package br.com.helpiguacu.backend.infrastructure.persistence.donation

import br.com.helpiguacu.backend.domain.model.donation.StatusDonation
import br.com.helpiguacu.backend.infrastructure.persistence.donationitem.DonationItem
import br.com.helpiguacu.backend.infrastructure.persistence.user.User
import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany
import jakarta.persistence.Table
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import java.time.LocalDateTime

@Entity
@Table(name = "donations")
class Donation(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)

    @Column(nullable = false)
    val id: Long? = null,

    @Column(nullable = false)
    val title: String,

    @Column(nullable = false)
    val description: String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    val donor: User,

    @Enumerated(EnumType.STRING)
    var status: StatusDonation = StatusDonation.PENDENTE_COLETA,

    @CreatedDate
    val createdAt: LocalDateTime = LocalDateTime.now(),

    @LastModifiedDate
    var updatedAt: LocalDateTime = LocalDateTime.now(),

    @OneToMany(
        mappedBy = "donation",
        cascade = [CascadeType.ALL],
        orphanRemoval = true
    )
    val items: MutableList<DonationItem> = mutableListOf()
)